File: qt6/gen_qkeysequence.cpp
#include <QKeyCombination>
#include <QKeySequence>
#include <QList>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qkeysequence.h>
#include "gen_qkeysequence.h"
#include "_cgo_export.h"

void QKeySequence_new(QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence();
	*outptr_QKeySequence = ret;
}

void QKeySequence_new2(struct miqt_string key, QKeySequence** outptr_QKeySequence) {
	QString key_QString = QString::fromUtf8(key.data, key.len);
	QKeySequence* ret = new QKeySequence(key_QString);
	*outptr_QKeySequence = ret;
}

void QKeySequence_new3(int k1, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(static_cast<int>(k1));
	*outptr_QKeySequence = ret;
}

void QKeySequence_new4(QKeyCombination* k1, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(*k1);
	*outptr_QKeySequence = ret;
}

void QKeySequence_new5(QKeySequence* ks, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(*ks);
	*outptr_QKeySequence = ret;
}

void QKeySequence_new6(int key, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(static_cast<QKeySequence::StandardKey>(key));
	*outptr_QKeySequence = ret;
}

void QKeySequence_new7(struct miqt_string key, int format, QKeySequence** outptr_QKeySequence) {
	QString key_QString = QString::fromUtf8(key.data, key.len);
	QKeySequence* ret = new QKeySequence(key_QString, static_cast<QKeySequence::SequenceFormat>(format));
	*outptr_QKeySequence = ret;
}

void QKeySequence_new8(int k1, int k2, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(static_cast<int>(k1), static_cast<int>(k2));
	*outptr_QKeySequence = ret;
}

void QKeySequence_new9(int k1, int k2, int k3, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(static_cast<int>(k1), static_cast<int>(k2), static_cast<int>(k3));
	*outptr_QKeySequence = ret;
}

void QKeySequence_new10(int k1, int k2, int k3, int k4, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(static_cast<int>(k1), static_cast<int>(k2), static_cast<int>(k3), static_cast<int>(k4));
	*outptr_QKeySequence = ret;
}

void QKeySequence_new11(QKeyCombination* k1, QKeyCombination* k2, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(*k1, *k2);
	*outptr_QKeySequence = ret;
}

void QKeySequence_new12(QKeyCombination* k1, QKeyCombination* k2, QKeyCombination* k3, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(*k1, *k2, *k3);
	*outptr_QKeySequence = ret;
}

void QKeySequence_new13(QKeyCombination* k1, QKeyCombination* k2, QKeyCombination* k3, QKeyCombination* k4, QKeySequence** outptr_QKeySequence) {
	QKeySequence* ret = new QKeySequence(*k1, *k2, *k3, *k4);
	*outptr_QKeySequence = ret;
}

int QKeySequence_Count(const QKeySequence* self) {
	return self->count();
}

bool QKeySequence_IsEmpty(const QKeySequence* self) {
	return self->isEmpty();
}

struct miqt_string QKeySequence_ToString(const QKeySequence* self) {
	QString _ret = self->toString();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

QKeySequence* QKeySequence_FromString(struct miqt_string str) {
	QString str_QString = QString::fromUtf8(str.data, str.len);
	return new QKeySequence(QKeySequence::fromString(str_QString));
}

struct miqt_array /* of QKeySequence* */  QKeySequence_ListFromString(struct miqt_string str) {
	QString str_QString = QString::fromUtf8(str.data, str.len);
	QList<QKeySequence> _ret = QKeySequence::listFromString(str_QString);
	// Convert QList<> from C++ memory to manually-managed C memory
	QKeySequence** _arr = static_cast<QKeySequence**>(malloc(sizeof(QKeySequence*) * _ret.length()));
	for (size_t i = 0, e = _ret.length(); i < e; ++i) {
		_arr[i] = new QKeySequence(_ret[i]);
	}
	struct miqt_array _out;
	_out.len = _ret.length();
	_out.data = static_cast<void*>(_arr);
	return _out;
}

struct miqt_string QKeySequence_ListToString(struct miqt_array /* of QKeySequence* */  list) {
	QList<QKeySequence> list_QList;
	list_QList.reserve(list.len);
	QKeySequence** list_arr = static_cast<QKeySequence**>(list.data);
	for(size_t i = 0; i < list.len; ++i) {
		list_QList.push_back(*(list_arr[i]));
	}
	QString _ret = QKeySequence::listToString(list_QList);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

int QKeySequence_Matches(const QKeySequence* self, QKeySequence* seq) {
	QKeySequence::SequenceMatch _ret = self->matches(*seq);
	return static_cast<int>(_ret);
}

QKeySequence* QKeySequence_Mnemonic(struct miqt_string text) {
	QString text_QString = QString::fromUtf8(text.data, text.len);
	return new QKeySequence(QKeySequence::mnemonic(text_QString));
}

struct miqt_array /* of QKeySequence* */  QKeySequence_KeyBindings(int key) {
	QList<QKeySequence> _ret = QKeySequence::keyBindings(static_cast<QKeySequence::StandardKey>(key));
	// Convert QList<> from C++ memory to manually-managed C memory
	QKeySequence** _arr = static_cast<QKeySequence**>(malloc(sizeof(QKeySequence*) * _ret.length()));
	for (size_t i = 0, e = _ret.length(); i < e; ++i) {
		_arr[i] = new QKeySequence(_ret[i]);
	}
	struct miqt_array _out;
	_out.len = _ret.length();
	_out.data = static_cast<void*>(_arr);
	return _out;
}

QKeyCombination* QKeySequence_OperatorSubscript(const QKeySequence* self, unsigned int i) {
	return new QKeyCombination(self->operator[](static_cast<uint>(i)));
}

void QKeySequence_OperatorAssign(QKeySequence* self, QKeySequence* other) {
	self->operator=(*other);
}

void QKeySequence_Swap(QKeySequence* self, QKeySequence* other) {
	self->swap(*other);
}

bool QKeySequence_OperatorEqual(const QKeySequence* self, QKeySequence* other) {
	return self->operator==(*other);
}

bool QKeySequence_OperatorNotEqual(const QKeySequence* self, QKeySequence* other) {
	return self->operator!=(*other);
}

bool QKeySequence_OperatorLesser(const QKeySequence* self, QKeySequence* ks) {
	return self->operator<(*ks);
}

bool QKeySequence_OperatorGreater(const QKeySequence* self, QKeySequence* other) {
	return self->operator>(*other);
}

bool QKeySequence_OperatorLesserOrEqual(const QKeySequence* self, QKeySequence* other) {
	return self->operator<=(*other);
}

bool QKeySequence_OperatorGreaterOrEqual(const QKeySequence* self, QKeySequence* other) {
	return self->operator>=(*other);
}

bool QKeySequence_IsDetached(const QKeySequence* self) {
	return self->isDetached();
}

struct miqt_string QKeySequence_ToString1(const QKeySequence* self, int format) {
	QString _ret = self->toString(static_cast<QKeySequence::SequenceFormat>(format));
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

QKeySequence* QKeySequence_FromString2(struct miqt_string str, int format) {
	QString str_QString = QString::fromUtf8(str.data, str.len);
	return new QKeySequence(QKeySequence::fromString(str_QString, static_cast<QKeySequence::SequenceFormat>(format)));
}

struct miqt_array /* of QKeySequence* */  QKeySequence_ListFromString2(struct miqt_string str, int format) {
	QString str_QString = QString::fromUtf8(str.data, str.len);
	QList<QKeySequence> _ret = QKeySequence::listFromString(str_QString, static_cast<QKeySequence::SequenceFormat>(format));
	// Convert QList<> from C++ memory to manually-managed C memory
	QKeySequence** _arr = static_cast<QKeySequence**>(malloc(sizeof(QKeySequence*) * _ret.length()));
	for (size_t i = 0, e = _ret.length(); i < e; ++i) {
		_arr[i] = new QKeySequence(_ret[i]);
	}
	struct miqt_array _out;
	_out.len = _ret.length();
	_out.data = static_cast<void*>(_arr);
	return _out;
}

struct miqt_string QKeySequence_ListToString2(struct miqt_array /* of QKeySequence* */  list, int format) {
	QList<QKeySequence> list_QList;
	list_QList.reserve(list.len);
	QKeySequence** list_arr = static_cast<QKeySequence**>(list.data);
	for(size_t i = 0; i < list.len; ++i) {
		list_QList.push_back(*(list_arr[i]));
	}
	QString _ret = QKeySequence::listToString(list_QList, static_cast<QKeySequence::SequenceFormat>(format));
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

void QKeySequence_Delete(QKeySequence* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QKeySequence*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qkeysequence.go
package qt

import (
	"unsafe"
)

type QKeySequence__StandardKey int

const (
	QKeySequence__UnknownKey               QKeySequence__StandardKey = 0
	QKeySequence__HelpContents             QKeySequence__StandardKey = 1
	QKeySequence__WhatsThis                QKeySequence__StandardKey = 2
	QKeySequence__Open                     QKeySequence__StandardKey = 3
	QKeySequence__Close                    QKeySequence__StandardKey = 4
	QKeySequence__Save                     QKeySequence__StandardKey = 5
	QKeySequence__New                      QKeySequence__StandardKey = 6
	QKeySequence__Delete                   QKeySequence__StandardKey = 7
	QKeySequence__Cut                      QKeySequence__StandardKey = 8
	QKeySequence__Copy                     QKeySequence__StandardKey = 9
	QKeySequence__Paste                    QKeySequence__StandardKey = 10
	QKeySequence__Undo                     QKeySequence__StandardKey = 11
	QKeySequence__Redo                     QKeySequence__StandardKey = 12
	QKeySequence__Back                     QKeySequence__StandardKey = 13
	QKeySequence__Forward                  QKeySequence__StandardKey = 14
	QKeySequence__Refresh                  QKeySequence__StandardKey = 15
	QKeySequence__ZoomIn                   QKeySequence__StandardKey = 16
	QKeySequence__ZoomOut                  QKeySequence__StandardKey = 17
	QKeySequence__Print                    QKeySequence__StandardKey = 18
	QKeySequence__AddTab                   QKeySequence__StandardKey = 19
	QKeySequence__NextChild                QKeySequence__StandardKey = 20
	QKeySequence__PreviousChild            QKeySequence__StandardKey = 21
	QKeySequence__Find                     QKeySequence__StandardKey = 22
	QKeySequence__FindNext                 QKeySequence__StandardKey = 23
	QKeySequence__FindPrevious             QKeySequence__StandardKey = 24
	QKeySequence__Replace                  QKeySequence__StandardKey = 25
	QKeySequence__SelectAll                QKeySequence__StandardKey = 26
	QKeySequence__Bold                     QKeySequence__StandardKey = 27
	QKeySequence__Italic                   QKeySequence__StandardKey = 28
	QKeySequence__Underline                QKeySequence__StandardKey = 29
	QKeySequence__MoveToNextChar           QKeySequence__StandardKey = 30
	QKeySequence__MoveToPreviousChar       QKeySequence__StandardKey = 31
	QKeySequence__MoveToNextWord           QKeySequence__StandardKey = 32
	QKeySequence__MoveToPreviousWord       QKeySequence__StandardKey = 33
	QKeySequence__MoveToNextLine           QKeySequence__StandardKey = 34
	QKeySequence__MoveToPreviousLine       QKeySequence__StandardKey = 35
	QKeySequence__MoveToNextPage           QKeySequence__StandardKey = 36
	QKeySequence__MoveToPreviousPage       QKeySequence__StandardKey = 37
	QKeySequence__MoveToStartOfLine        QKeySequence__StandardKey = 38
	QKeySequence__MoveToEndOfLine          QKeySequence__StandardKey = 39
	QKeySequence__MoveToStartOfBlock       QKeySequence__StandardKey = 40
	QKeySequence__MoveToEndOfBlock         QKeySequence__StandardKey = 41
	QKeySequence__MoveToStartOfDocument    QKeySequence__StandardKey = 42
	QKeySequence__MoveToEndOfDocument      QKeySequence__StandardKey = 43
	QKeySequence__SelectNextChar           QKeySequence__StandardKey = 44
	QKeySequence__SelectPreviousChar       QKeySequence__StandardKey = 45
	QKeySequence__SelectNextWord           QKeySequence__StandardKey = 46
	QKeySequence__SelectPreviousWord       QKeySequence__StandardKey = 47
	QKeySequence__SelectNextLine           QKeySequence__StandardKey = 48
	QKeySequence__SelectPreviousLine       QKeySequence__StandardKey = 49
	QKeySequence__SelectNextPage           QKeySequence__StandardKey = 50
	QKeySequence__SelectPreviousPage       QKeySequence__StandardKey = 51
	QKeySequence__SelectStartOfLine        QKeySequence__StandardKey = 52
	QKeySequence__SelectEndOfLine          QKeySequence__StandardKey = 53
	QKeySequence__SelectStartOfBlock       QKeySequence__StandardKey = 54
	QKeySequence__SelectEndOfBlock         QKeySequence__StandardKey = 55
	QKeySequence__SelectStartOfDocument    QKeySequence__StandardKey = 56
	QKeySequence__SelectEndOfDocument      QKeySequence__StandardKey = 57
	QKeySequence__DeleteStartOfWord        QKeySequence__StandardKey = 58
	QKeySequence__DeleteEndOfWord          QKeySequence__StandardKey = 59
	QKeySequence__DeleteEndOfLine          QKeySequence__StandardKey = 60
	QKeySequence__InsertParagraphSeparator QKeySequence__StandardKey = 61
	QKeySequence__InsertLineSeparator      QKeySequence__StandardKey = 62
	QKeySequence__SaveAs                   QKeySequence__StandardKey = 63
	QKeySequence__Preferences              QKeySequence__StandardKey = 64
	QKeySequence__Quit                     QKeySequence__StandardKey = 65
	QKeySequence__FullScreen               QKeySequence__StandardKey = 66
	QKeySequence__Deselect                 QKeySequence__StandardKey = 67
	QKeySequence__DeleteCompleteLine       QKeySequence__StandardKey = 68
	QKeySequence__Backspace                QKeySequence__StandardKey = 69
	QKeySequence__Cancel                   QKeySequence__StandardKey = 70
)

type QKeySequence__SequenceFormat int

const (
	QKeySequence__NativeText   QKeySequence__SequenceFormat = 0
	QKeySequence__PortableText QKeySequence__SequenceFormat = 1
)

type QKeySequence__SequenceMatch int

const (
	QKeySequence__NoMatch      QKeySequence__SequenceMatch = 0
	QKeySequence__PartialMatch QKeySequence__SequenceMatch = 1
	QKeySequence__ExactMatch   QKeySequence__SequenceMatch = 2
)

type QKeySequence struct {
	h          uintptr
	isSubclass bool
}

// NewQKeySequence constructs a new QKeySequence object.
func NewQKeySequence() *QKeySequence {

	ret := newQKeySequence(QKeySequence_new())
	ret.isSubclass = true
	return ret
}

// NewQKeySequence2 constructs a new QKeySequence object.
func NewQKeySequence2(key string) *QKeySequence {
	key_ms := struct_miqt_string{}
	key_ms.data = CString(key)
	key_ms.len = size_t(len(key))
	defer free(unsafe.Pointer(key_ms.data))

	ret := newQKeySequence(QKeySequence_new2(key_ms))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence3 constructs a new QKeySequence object.
func NewQKeySequence3(k1 int) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new3((int)(k1)))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence4 constructs a new QKeySequence object.
func NewQKeySequence4(k1 QKeyCombination) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new4(k1.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence5 constructs a new QKeySequence object.
func NewQKeySequence5(ks *QKeySequence) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new5(ks.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence6 constructs a new QKeySequence object.
func NewQKeySequence6(key StandardKey) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new6(key))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence7 constructs a new QKeySequence object.
func NewQKeySequence7(key string, format SequenceFormat) *QKeySequence {
	key_ms := struct_miqt_string{}
	key_ms.data = CString(key)
	key_ms.len = size_t(len(key))
	defer free(unsafe.Pointer(key_ms.data))

	ret := newQKeySequence(QKeySequence_new7(key_ms, format))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence8 constructs a new QKeySequence object.
func NewQKeySequence8(k1 int, k2 int) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new8((int)(k1), (int)(k2)))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence9 constructs a new QKeySequence object.
func NewQKeySequence9(k1 int, k2 int, k3 int) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new9((int)(k1), (int)(k2), (int)(k3)))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence10 constructs a new QKeySequence object.
func NewQKeySequence10(k1 int, k2 int, k3 int, k4 int) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new10((int)(k1), (int)(k2), (int)(k3), (int)(k4)))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence11 constructs a new QKeySequence object.
func NewQKeySequence11(k1 QKeyCombination, k2 QKeyCombination) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new11(k1.cPointer(), k2.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence12 constructs a new QKeySequence object.
func NewQKeySequence12(k1 QKeyCombination, k2 QKeyCombination, k3 QKeyCombination) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new12(k1.cPointer(), k2.cPointer(), k3.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQKeySequence13 constructs a new QKeySequence object.
func NewQKeySequence13(k1 QKeyCombination, k2 QKeyCombination, k3 QKeyCombination, k4 QKeyCombination) *QKeySequence {

	ret := newQKeySequence(QKeySequence_new13(k1.cPointer(), k2.cPointer(), k3.cPointer(), k4.cPointer()))
	ret.isSubclass = true
	return ret
}

func (this *QKeySequence) Count() int {
	return (int)(QKeySequence_Count(this.h))
}

func (this *QKeySequence) IsEmpty() bool {
	return (bool)(QKeySequence_IsEmpty(this.h))
}

func (this *QKeySequence) ToString() string {
	var _ms struct_miqt_string = QKeySequence_ToString(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QKeySequence_FromString(str string) *QKeySequence {
	str_ms := struct_miqt_string{}
	str_ms.data = CString(str)
	str_ms.len = size_t(len(str))
	defer free(unsafe.Pointer(str_ms.data))
	_goptr := newQKeySequence(QKeySequence_FromString(str_ms))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func QKeySequence_ListFromString(str string) []QKeySequence {
	str_ms := struct_miqt_string{}
	str_ms.data = CString(str)
	str_ms.len = size_t(len(str))
	defer free(unsafe.Pointer(str_ms.data))
	var _ma struct_miqt_array = QKeySequence_ListFromString(str_ms)
	_ret := make([]QKeySequence, int(_ma.len))
	_outCast := (*[0xffff]*QKeySequence)(unsafe.Pointer(_ma.data)) // hey ya
	for i := 0; i < int(_ma.len); i++ {
		_lv_goptr := newQKeySequence(_outCast[i])
		_lv_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
		_ret[i] = *_lv_goptr
	}
	return _ret
}

func QKeySequence_ListToString(list []QKeySequence) string {
	list_CArray := (*[0xffff]*QKeySequence)(malloc(size_t(8 * len(list))))
	defer free(unsafe.Pointer(list_CArray))
	for i := range list {
		list_CArray[i] = list[i].cPointer()
	}
	list_ma := struct_miqt_array{len: size_t(len(list)), data: unsafe.Pointer(list_CArray)}
	var _ms struct_miqt_string = QKeySequence_ListToString(list_ma)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QKeySequence) Matches(seq *QKeySequence) SequenceMatch {
	xxxxxxxxx
}

func QKeySequence_Mnemonic(text string) *QKeySequence {
	text_ms := struct_miqt_string{}
	text_ms.data = CString(text)
	text_ms.len = size_t(len(text))
	defer free(unsafe.Pointer(text_ms.data))
	_goptr := newQKeySequence(QKeySequence_Mnemonic(text_ms))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func QKeySequence_KeyBindings(key StandardKey) []QKeySequence {
	var _ma struct_miqt_array = QKeySequence_KeyBindings(key)
	_ret := make([]QKeySequence, int(_ma.len))
	_outCast := (*[0xffff]*QKeySequence)(unsafe.Pointer(_ma.data)) // hey ya
	for i := 0; i < int(_ma.len); i++ {
		_lv_goptr := newQKeySequence(_outCast[i])
		_lv_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
		_ret[i] = *_lv_goptr
	}
	return _ret
}

func (this *QKeySequence) OperatorSubscript(i uint) *QKeyCombination {
	_goptr := newQKeyCombination(QKeySequence_OperatorSubscript(this.h, (uint)(i)))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QKeySequence) OperatorAssign(other *QKeySequence) {
	QKeySequence_OperatorAssign(this.h, other.cPointer())
}

func (this *QKeySequence) Swap(other *QKeySequence) {
	QKeySequence_Swap(this.h, other.cPointer())
}

func (this *QKeySequence) OperatorEqual(other *QKeySequence) bool {
	return (bool)(QKeySequence_OperatorEqual(this.h, other.cPointer()))
}

func (this *QKeySequence) OperatorNotEqual(other *QKeySequence) bool {
	return (bool)(QKeySequence_OperatorNotEqual(this.h, other.cPointer()))
}

func (this *QKeySequence) OperatorLesser(ks *QKeySequence) bool {
	return (bool)(QKeySequence_OperatorLesser(this.h, ks.cPointer()))
}

func (this *QKeySequence) OperatorGreater(other *QKeySequence) bool {
	return (bool)(QKeySequence_OperatorGreater(this.h, other.cPointer()))
}

func (this *QKeySequence) OperatorLesserOrEqual(other *QKeySequence) bool {
	return (bool)(QKeySequence_OperatorLesserOrEqual(this.h, other.cPointer()))
}

func (this *QKeySequence) OperatorGreaterOrEqual(other *QKeySequence) bool {
	return (bool)(QKeySequence_OperatorGreaterOrEqual(this.h, other.cPointer()))
}

func (this *QKeySequence) IsDetached() bool {
	return (bool)(QKeySequence_IsDetached(this.h))
}

func (this *QKeySequence) DataPtr() *DataPtr {
	xxxxxxxxx
}

func (this *QKeySequence) ToString1(format SequenceFormat) string {
	var _ms struct_miqt_string = QKeySequence_ToString1(this.h, format)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QKeySequence_FromString2(str string, format SequenceFormat) *QKeySequence {
	str_ms := struct_miqt_string{}
	str_ms.data = CString(str)
	str_ms.len = size_t(len(str))
	defer free(unsafe.Pointer(str_ms.data))
	_goptr := newQKeySequence(QKeySequence_FromString2(str_ms, format))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func QKeySequence_ListFromString2(str string, format SequenceFormat) []QKeySequence {
	str_ms := struct_miqt_string{}
	str_ms.data = CString(str)
	str_ms.len = size_t(len(str))
	defer free(unsafe.Pointer(str_ms.data))
	var _ma struct_miqt_array = QKeySequence_ListFromString2(str_ms, format)
	_ret := make([]QKeySequence, int(_ma.len))
	_outCast := (*[0xffff]*QKeySequence)(unsafe.Pointer(_ma.data)) // hey ya
	for i := 0; i < int(_ma.len); i++ {
		_lv_goptr := newQKeySequence(_outCast[i])
		_lv_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
		_ret[i] = *_lv_goptr
	}
	return _ret
}

func QKeySequence_ListToString2(list []QKeySequence, format SequenceFormat) string {
	list_CArray := (*[0xffff]*QKeySequence)(malloc(size_t(8 * len(list))))
	defer free(unsafe.Pointer(list_CArray))
	for i := range list {
		list_CArray[i] = list[i].cPointer()
	}
	list_ma := struct_miqt_array{len: size_t(len(list)), data: unsafe.Pointer(list_CArray)}
	var _ms struct_miqt_string = QKeySequence_ListToString2(list_ma, format)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}
